pkg/api/controller.go

```diff
@@ -1776,6 +1778,9 @@ func (c *Controller) handleAPIError(ctx context.Context, w http.ResponseWriter,
 	case errors.Is(err, graveler.ErrTooManyTries):
 		writeError(w, http.StatusLocked, "Too many attempts, try again later")
 
+	case errors.Is(err, context.Canceled):
```

Comment:
Just realized something. We should differentiate client and server side cancellation errors. To return `HTTPStatusRequestCanceled` we need to check that `request.Context` was cancelled. Checking the return error for the same might cover all requests cancellation, but it might also cover other reasons for cancellations, some of them are related lakeFS logic, or external components (object store, DB, ...)

Reply:
@itaiad200 - this was actually brought up on the original issue, where you claimed (rightfuly, IMO) that a cancellation might also come from the server side (our logic, or other component).
I currently introduced a general "Request Canceled" without refering to which side the cancellation comes from.
If you want to distinguish the 2, it will require another new code.
How significant is the distinguish between the two? If I understand correctly, if the client cancels the request, it is not really interested in the result, as it is aware of the cancellation. If the client is actually receiving an unexpected "Request Canceled" it can only come from the server side (right? am I missing something?)

Reply:
Right - if we cancelled the request it should not return `499. `500` is more appropriate. We need to check if the http request context was cancelled and return `499` only if it was.

Reply:
I've checked the request's context for cancellation for this reason.
About the server side cancel concern - if we are using context.WithTimeout the error will be context deadline exceeded. So unless we have a specific code that cancel the context in the request level it should be same to assume that the cancellation is by the http handler on client closed request.

Reply:
Not sure I follow @nopcoder. If the context was cancelled by some downstream component, like graveler or pgx client, then the context on the request will not be cancelled and the status code we return to the client isn't 499. The only reason we'll return 499 is if the `request.Context` was cancelled, and we don't control that, right?